src/helpers/SelectedCharacteristicsContext.jsx

```diff
@@ -0,0 +1,39 @@
+import { createContext, useState } from 'react';
+
+export const CharacteristicsContext = createContext();
+
+export function SelectedCharacteristicsContext(props) {
+  const [selectedCharacteristics, setSelectedCharacteristics] = useState([]);
+
+  const handleSelect = (characteristics) => {
+    if (selectedCharacteristics.includes(characteristics)) {
+      return;
+    }
+    setSelectedCharacteristics((selectedCharacteristics) => [
+      ...selectedCharacteristics,
+      characteristics,
+    ]);
+  };
+
+  const handleRemove = (characteristics) => {
+    setSelectedCharacteristics((selectedCharacteristics) =>
+      selectedCharacteristics.splice(selectedCharacteristics.indexOf(characteristics), 1),
```

Comment:
Splice method isn't the best option here. It mutates the original array. Try to find an immutable approach.